ui/src/pages/Alert/Channel/ChannelDrawer.tsx

```diff
@@ -0,0 +1,164 @@
+import { alert } from '@/api';
+import AlertDrawer from '@/components/AlertDrawer';
+import { CHANNEL_TYPE_OPTIONS } from '@/constants';
+import { QuestionCircleOutlined } from '@ant-design/icons';
+import { useRequest } from 'ahooks';
+import type { DrawerProps } from 'antd';
+import { Button, Form, Input, Select, Space } from 'antd';
+import { useEffect } from 'react';
+
+/**
+ * ChannelDrawer has three states: creation,editing and pure display
+ * Editing state and pure display state can be switched between each other
+ */
+interface ChannelDrawerProps extends DrawerProps {
+  status: Alert.DrawerStatus;
+  name?: string;
+  setStatus?: (status: Alert.DrawerStatus) => void;
+}
+
+const { TextArea } = Input;
+
+export default function ChannelDrawer({
+  status,
+  name,
+  onClose,
+  setStatus,
+  ...props
+}: ChannelDrawerProps) {
+  const [form] = Form.useForm();
+  const { run: getReceiver } = useRequest(alert.getReceiver, {
+    manual: true,
+    onSuccess: ({ successful, data }) => {
+      if (successful) {
+        form.setFieldsValue({
+          ...data,
+        });
+      }
+    },
+  });
+  const { data: listReceiversRes } = useRequest(alert.listReceivers, {
+    manual: true,
+  });
+  const receiverNames = listReceiversRes?.data.map((receiver) => receiver.name);
+
+  const Footer = () => {
+    return (
+      <div>
+        <Space>
+          <Button
+            onClick={
+              status === 'display' && setStatus
+                ? () => setStatus('edit')
+                : () => form.submit()
+            }
+            type="primary"
+          >
+            {status === 'display' ? '编辑' : '提交'}
+          </Button>
+          <Button onClick={onClose}>取消</Button>
+        </Space>
+      </div>
+    );
+  };
+
+  useEffect(() => {
+    if (status !== 'create' && name) {
+      getReceiver(name);
+    }
+  }, [status]);
```

Comment:
Why not put the name in the dependency list? I thought it's a dependency in this effect, too.